cmd/brew-portable-package.rb

```diff
@@ -12,6 +12,7 @@
 #:    A new commit will then be generated unless `--no-commit` is passed.
 
 ENV["HOMEBREW_DEVELOPER"] = "1"
+ENV["HOMEBREW_BUILD_BOTTLE"] = "1" if OS.linux?
```

Comment:
Why is this needed? If it's for `safe_system "brew", "install", "-s", *deps` can you change that to `--build-bottle` instead?

Reply:
Coming to think about it, I think all portable formulae should be built with `--build-bottle` flag. This is required to ensure portable formulae are built with correct `-march` flag. Moreover, `--build-bottle` or `HOMEBREW_BUILD_BOTTLE` only affect current formula instead of its dependencies.

So I think we should change it to something like this:
```ruby
    deps = Utils.popen_read("brew", "deps", "--include-build", name).split("\n")

    graph = {}
    deps.each { |dep| graph[dep] = Formula[dep].deps.map(&:name) }
    deps = TSort.tsort ->(&b) { graph.each_key(&b) }, ->(n, &b) { graph[n].each(&b) }

    portable_deps = deps.select { |dep| dep.start_with? "portable-" }
    other_deps = deps - portable_deps

    # Avoid installing glibc. Bottles depend on glibc.
    other_deps.each { |dep| safe_system "brew", "install", "-s", dep } if OS.linux?

    portable_deps.each { |dep| safe_system "brew", "install", "--build-bottle", dep }
    safe_system "brew", "install", "--build-bottle", name
```

If it's OK to you, I can send a separated PR for this.

Reply:
`--build-bottle` will use the same `march` on macOS, FYI. The main issue is that it will never run `post_install` (which shouldn't be an issue here but is worth being explicit).

I don't think it's necessary to do more than just use `--build-bottle` instead of `-s`. We don't need to build e.g. `pkg-config` as a bottle.

Reply:
> --build-bottle will use the same march on macOS, FYI.

I don't think this is true for linux.

> I don't think it's necessary to do more than just use --build-bottle instead of -s. We don't need to build e.g. pkg-config as a bottle.

In the above code, `--build-bottle` is only passed to portable formulae.

Reply:
> In the above code, `--build-bottle` is only passed to portable formulae.

If the others are being built from source anyway (on Linux): I don't think it matters. I'm open to changing the existing `-s` to `--build-bottle` on Linux but I don't see a need in going beyond that.

Let's not make changes that aren't demonstrated to be necessary through e.g. reproducible bugs.

Reply:
> Let's not make changes that aren't demonstrated to be necessary through e.g. reproducible bugs.

The reason that there is no bug for now is because we currently set `HOMEBREW_BUILD_BOTTLE` in the docker. This is effectively equivalent to passing `--build-bottle` to all formulae on Linux. If we remove `HOMEBREW_BUILD_BOTTLE`, the portable ruby won't work on old CPU.

> If the others are being built from source anyway (on Linux): I don't think it matters. I'm open to changing the existing -s to --build-bottle on Linux but I don't see a need in going beyond that.

This is not enough. `--build-bottle` flag only affects the current formula not its deps.

Reply:
Ok, leave this as-is.